src/store/normalize-data.js

```diff
@@ -239,8 +239,10 @@ const normalizeData = (data, expandAllPipelines) => {
         }
       });
     } else {
-      for (const child of data.modular_pipelines['__root__'].children) {
-        state.modularPipeline.visible[child.id] = true;
+      if (data.modular_pipelines && data.modular_pipelines['__root__']) {
```

Comment:
can you explain if this was a bug fix? :)

Reply:
I am sure this will never happen but, when I passed an empty json response from fixture emptyDataset.json which might not have a proper schema (i.e., the backend response schema) this caused an exception in the application. To handle it gracefully, I added the check. We can adjust the empty schema and get rid of this change. But I felt it is better to handle this way. Let me know if you feel otherwise